### PR TITLE
custom LooseVersion replaces deprecated distutils.version.LooseVersion

### PR DESCRIPTION
**Pull request by iamkaushal**
resolved issue #316 by creating a similar LooseVersion class for the deprecated distutils.version.LooseVersion.

### CODE REVIEW

File: openeo/capabilities.py [hunk 1]
```diff
@@ -1,10 +1,92 @@
...
 from abc import ABC
-from distutils.version import LooseVersion
+import re
 from typing import Union
 
 # Is this base class (still) useful?
 
 
+class LooseVersion:
+    """A version number consists of a series of numbers,
```
Comment (soxofaan) [inline openeo/capabilities.py:10]: If I understand correctly this is a straight copy from the original `distutils.version.LooseVersion`.
I think that should be noted in the docstring.

Moreover it now looks unnecessary and overkill to have both LooseVersion and ComparableVersion defined here. I think it makes more sense to merge both classes to a single class

File: openeo/capabilities.py [hunk 2]
```diff
@@ -1,10 +1,92 @@
...
+    """
+
+    component_re = re.compile(r'(\d+ | [a-z]+ | \.)', re.VERBOSE)
+
+    def __init__(self, version_string=None):
+        if isinstance(version_string, str):
+            self.parse(version_string)
+
+        if isinstance(version_string, LooseVersion):
+            self.version = version_string.version
```
Comment (soxofaan) [inline openeo/capabilities.py:42]: there is no `else` here: what if `version_string=None` (the default)?

File: openeo/capabilities.py [hunk 3]
```diff
@@ -1,10 +1,92 @@
...
+
+    def parse(self, version_string):
+        self.vstring = version_string
```
Comment (soxofaan) [inline openeo/capabilities.py:45]: this method introduces `self.vstring` which is not defined in `__init__`

**Review by soxofaan — COMMENTED** [review]
Thanks for this PR!
Sorry for the delay in reviewing this (I had to work through my backlog after my summer holidays)

File: openeo/capabilities.py (hunk 2 above)
Comment (iamkaushal) [reply openeo/capabilities.py]: fixed

File: openeo/capabilities.py (hunk 3 above)
Comment (iamkaushal) [reply openeo/capabilities.py]: fixed

File: openeo/capabilities.py (hunk 1 above)
Comment (iamkaushal) [reply openeo/capabilities.py]: Makes sense. Fixed.

**Comment by iamkaushal** [comment]
Thank you for the review, I have made the changes, please review again.

File: openeo/capabilities.py [hunk 4]
```diff
@@ -1,9 +1,8 @@
+import contextlib
 from abc import ABC
-from distutils.version import LooseVersion
+import re
 from typing import Union
 
-# Is this base class (still) useful?
```
Comment (soxofaan) [inline openeo/capabilities.py:5]: This comment shouldn't be removed (it applies to Capabilities class)

File: openeo/capabilities.py [hunk 5]
```diff
@@ -52,7 +51,33 @@ def list_plans(self):
 
 class ComparableVersion:
     """
-    Helper to compare a version (e.g. API version) against another (threshold) version
+    This code is copied from distutils.version.LooseVersion
```
Comment (soxofaan) [inline openeo/capabilities.py:54]: Now after being merged, I would not say "This code is copied", but "implementation is based on"

File: openeo/capabilities.py [hunk 6]
```diff
@@ -78,11 +103,21 @@ class ComparableVersion:
         `a.accept_lower(b)`: b is lower than a
     """
 
+    component_re = re.compile(r'(\d+ | [a-z]+ | \.)', re.VERBOSE)
```
Comment (soxofaan) [inline openeo/capabilities.py:106]: I would make this a private attribute

File: openeo/capabilities.py [hunk 7]
```diff
@@ -78,11 +103,21 @@ class ComparableVersion:
...
+
     def __init__(self, version: Union[str, 'ComparableVersion']):
         if isinstance(version, ComparableVersion):
             self._version = version._version
         else:
-            self._version = LooseVersion(version)
+            self.parse(version)
+
+    def parse(self, version_string):
```
Comment (soxofaan) [inline openeo/capabilities.py:114]: I think this should be a (private) class method, that doesn't modify `self`, but just returns a tuple

**Review by soxofaan — COMMENTED** [review]
Thanks for the updates!

some more fine-tune suggestions

File: openeo/capabilities.py (hunk 5 above)
Comment (iamkaushal) [reply openeo/capabilities.py]: done

File: openeo/capabilities.py (hunk 6 above)
Comment (iamkaushal) [reply openeo/capabilities.py]: fixed

**Comment by soxofaan** [comment]
Thanks!

I did some final fine-tuning and merged in 17cbe08

**Comment by iamkaushal** [comment]
@soxofaan  Awesome, it was my first open source contribution, thank you very much for helping me throughout the process.

**Pull request closed without merging**